File: examples-sdk/swift/301-rag-basic-vector-search/main.swift
// 301-rag-basic-vector-search — Swift SDK equivalent
// Compile: vil compile --from swift --input 301-rag-basic-vector-search/main.swift --release

let server = VilServer(name: "rag-basic-vector-search", port: 3110)
let rag_basic = ServiceProcess(name: "rag-basic")
rag_basic.endpoint(method: "POST", path: "/rag", handler: "rag_handler")
server.service(rag_basic)
server.compile()

File: examples-sdk/csharp/301-rag-basic-vector-search/Main.cs
// 301-rag-basic-vector-search — C# SDK equivalent
// Compile: vil compile --from csharp --input 301-rag-basic-vector-search/Main.cs --release

#r "sdk/csharp/Vil.cs"

var server = new VilServer("rag-basic-vector-search", 3110);
var rag_basic = new ServiceProcess("rag-basic");
rag_basic.Endpoint("POST", "/rag", "rag_handler");
server.Service(rag_basic);
server.Compile();
